### ISSUE TITLE
Internal error when docstring end follows double quote

### ISSUE THREAD

**Issue by hugovk** [psf/black#1452]
**Describe the bug** A clear and concise description of what the bug is.

**To Reproduce** Steps to reproduce the behavior:

1. Take this file:

```python
def thing():
    """Controls vertical alignment of fields
    Arguments:

    valign - vertical alignment, one of "t", "m", or "b" """
    return
```

(Simplified from https://github.com/jazzband/prettytable/blob/master/prettytable.py)

2. Run _Black_ on it with these arguments: `black 1.py`
3. See error:
```console
$ black 1.py
error: cannot format 1.py: INTERNAL ERROR: Black produced invalid code: EOL while scanning string literal (<unknown>, line 5). Please report a bug on https://github.com/psf/black/issues.  This invalid output might be helpful: /var/folders/kt/j77sf4_n6fnbx6pg199rbx700000gn/T/blk_vs1w_3yc.log
Oh no! 💥 💔 💥
1 file failed to reformat.
```
```console
$ cat /var/folders/kt/j77sf4_n6fnbx6pg199rbx700000gn/T/blk_vs1w_3yc.log
  File "/Users/hugo/github/black/src/black/__init__.py", line 6037, in assert_equivalent
    dst_ast = parse_ast(dst)
  File "/Users/hugo/github/black/src/black/__init__.py", line 5948, in parse_ast
    return ast27.parse(src)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/typed_ast/ast27.py", line 50, in parse
    return _ast27.parse(source, filename, mode)
def thing():
    """Controls vertical alignment of fields
    Arguments:

    valign - vertical alignment, one of "t", "m", or "b""""
    return
```

**Expected behavior** A clear and concise description of what you expected to happen.

File was formatted.

**Environment (please complete the following information):**

- Version: master
- OS and Python version: macOS/Python 3.8.3.

**Does this bug also happen on master?** To answer this, you have two options:

Yes with `master`. No with latest 19.10b0 release, which does not change the file.

**Additional context** Add any other context about the problem here.

It's fine with single quotes instead in `"t", "m", or "b"`, or with a newline after `"b"`.

**Comment by hugovk** [comment]
Git bisect points to a4c11a75e12300abfbe4c36854e450d42bdd1ee7 as the first bad commit, from PR https://github.com/psf/black/pull/1053, "Re-indent the contents of docstrings".

PR https://github.com/psf/black/pull/1417 fixed some other docstring things (https://github.com/psf/black/issues/1415) but not this case. 

Here's a smaller failing case.

```python
def containing_quotes():
    """No quotes here

    "quotes here" """
    pass
```

**Comment by ambv** [comment]
Fixed by #1623.
